### PR TITLE
RFC30/Set fields to fluent builder

### PR DESCRIPTION
**Pull request by thomas-k-cameron**
## Motivation and Context
Part of RFC30

## Description
Add `fn set_fields` to fluent builder.

## Testing
NA

## Checklist
NA
----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt [hunk 1]
```diff
@@ -287,6 +287,14 @@ class FluentClientGenerator(
...
                         }
 
+                        $cfgAttribute
+                        /// This function replaces the parameter with new one.
+                        /// It is useful when you want to replace the existing data with de-serialized data.
+                        pub fn set_fields(mut self, data: #{InputBuilderType}) -> Self {
+                            self.inner = data;
+                            self
+                        }
+
```
[the remark this answers is not in the document]
Comment (thomas-k-cameron) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt:297]: `#{InputBuilderType}` is the builder type of the operation.
So, if the fluent builder you are building is S3's `aws_sdk_s3::client::fluent_builders::CopyObject` then the builder type would be `crate::input::copy_object_input::Builder`.

generated code would look like this.
```rust
    pub struct CopyObject {
        handle: std::sync::Arc<super::Handle>,
        inner: crate::input::copy_object_input::Builder,
    }
    impl CopyObject {
        /// Creates a new `CopyObject`.
        pub(crate) fn new(handle: std::sync::Arc<super::Handle>) -> Self {
            Self {
                handle,
                inner: Default::default(),
            }
        }

         #[cfg(aws_sdk_unstable)]
        /// This function replaces the parameter with new one.
        /// It is useful when you want to replace the existing data with de-serialized data.
        pub fn set_fields(
            mut self,
            data: crate::input::copy_object_input::Builder,
        ) -> Self {
            self.inner = data;
            self
        }
```

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt [hunk 2]
```diff
@@ -264,7 +264,7 @@ class FluentClientGenerator(
                 ) {
                     val outputType = symbolProvider.toSymbol(operation.outputShape(model))
                     val errorType = operation.errorSymbol(symbolProvider)
-
+                    val cfgAttribute = "##[cfg(aws_sdk_unstable)]"
```
[the remark this answers is not in the document]
Comment (thomas-k-cameron) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt]: Fixed

**Comment by thomas-k-cameron** [comment]
there is a bug. let me fix it up

**Comment by thomas-k-cameron** [comment]
@Velfi 
Would you kindly review this one?

**Review by Velfi — APPROVED** [review]

**Pull request merged**